mypy/checker.py

```diff
@@ -2358,26 +2388,60 @@ def find_isinstance_check(node: Node,
 
     Guaranteed to not return None, None. (But may return {}, {})
     """
+    def split_types(current_type: Optional[Type], type_if_true: Optional[Type], expr: Node) \
+            -> Tuple[Optional[Dict[Node, Type]], Optional[Dict[Node, Type]]]:
+        if type_if_true:
```

Comment:
This function probably deserves its own docstring (it was easier to follow what was going on from context in the old version).